ktor-shared/ktor-serialization/ktor-serialization-kotlinx/ktor-serialization-kotlinx-json/common/src/io/ktor/serialization/kotlinx/json/KotlinxSerializationJsonExtensions.kt

```diff
@@ -75,15 +75,22 @@ internal class KotlinxSerializationJsonExtensions(private val format: Json) : Ko
         charset: Charset,
         channel: ByteWriteChannel
     ) {
-        val jsonArraySymbols = jsonArraySymbolsMap.getOrPut(charset) { JsonArraySymbols(charset) }
+        // Use Big Endian charsets without byte-order mark (BOM)
+        val charsetWithoutBom = when(charset) {
+            Charsets.UTF_16 -> Charsets.UTF_16BE
```

Comment:
I don't think that these charsets are supported on native =(

Reply:
Understood, I removed this commit !